src/elm/Pages/Build/Update.elm

```diff
@@ -25,220 +23,117 @@ import Util exposing (overwriteById)
-                    , True
-                    )
+clickResource : WebData (Resources a) -> String -> ( WebData (Resources a), Bool )
+clickResource resources stepNumber =
+    resources
+        |> RemoteData.unwrap ( resources, False )
+            (\resources_ ->
+                ( toggleView stepNumber resources_ |> RemoteData.succeed
+                , True
                 )
-                steps
-    in
-    ( stepsOut
-    , action
-    )
+            )
 
 
-{-| mergeSteps : takes takes current steps and incoming step information and merges them, updating old logs and retaining previous state.
+{-| merge : takes takes current resources and incoming resource information and merges them, updating old logs and retaining previous state.
 -}
-mergeSteps : Maybe String -> Bool -> WebData Steps -> Steps -> Steps
-mergeSteps logFocus refresh currentSteps incomingSteps =
+merge : Maybe String -> Bool -> WebData (Resources a) -> Resources a -> Resources a
+merge logFocus refresh current incoming =
     let
-        updatedSteps =
-            currentSteps
-                |> RemoteData.unwrap incomingSteps
-                    (\steps ->
-                        incomingSteps
+        merged =
+            current
+                |> RemoteData.unwrap incoming
+                    (\resources ->
+                        incoming
                             |> List.map
-                                (\incomingStep ->
+                                (\r ->
                                     let
                                         ( viewing, focus ) =
-                                            getStepInfo steps incomingStep.number
+                                            getInfo resources r.number
+
+                                        s =
+                                            { r
+                                                | viewing = viewing
+                                                , logFocus = focus
+                                            }
                                     in
-                                    overwriteById
-                                        { incomingStep
-                                            | viewing = viewing
-                                            , logFocus = focus
-                                        }
-                                        steps
+                                    Just <| Maybe.withDefault s <| overwriteById s resources
                                 )
                             |> List.filterMap identity
                     )
     in
     -- when not an automatic refresh, respect the url focus
     if not refresh then
-        focusStep logFocus updatedSteps
+        focus logFocus merged
 
     else
-        updatedSteps
+        merged
 
 
-{-| isViewingStep : takes steps and step number and returns the step viewing state
+{-| isViewing : takes resources and resource number and returns the resource viewing state
```

Comment:
generalizing these functions so that they may be used for services in the next PR